doc-tests/lib/find_changes.rb

```diff
@@ -0,0 +1,60 @@
+# This function parses the YAML file in <gitroot>/deploy/kubernetes/complete-demo.yaml,
+# and extracts all used images.
+def find_images_in_k8s_complete_demo
+  images = []
+  yaml_parts = File.read(File.expand_path("../../../../deploy/kubernetes/complete-demo.yaml", __FILE__))\
+    .split("---\n")\
+    .map { |part| YAML.load(part) }
+
+  yaml_parts \
+    .select { |part| part["kind"] == "Deployment" }\
+    .map { |part| part["spec"]["template"]["spec"]["containers"].map { |c| c["image"] } }.flatten\
+    .map { |name| if name.include?(":") then name; else name + ":latest"; end }.uniq
+end
+
+# Checks if any of the +image_list+ images on the docker hub have been changed since +since+.
+def any_image_changed_since?(image_list, since)
+  log(:info, "Checking if any of #{image_list.join(", ")} has changed since #{since.rfc2822}")
+
+  changed_images = image_list.select do |img|
+    log_system_or_die(:debug, "docker pull #{img}")
```

Comment:
Ok, not going as fast as I had hoped.

Stuck at
```
[maarten@maarten-laptop:~]$ curl -v -I https://hub.docker.com/v2/repositories/library/mongo/manifests/3
*   Trying 52.71.230.17...
* TCP_NODELAY set
* Connected to hub.docker.com (52.71.230.17) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: OU=GT68254618; OU=See www.rapidssl.com/resources/cps (c)13; OU=Domain Control Validated - RapidSSL(R); CN=*.docker.com
*  start date: Oct 29 21:14:42 2014 GMT
*  expire date: Nov 13 14:19:33 2017 GMT
*  subjectAltName: host "hub.docker.com" matched cert's "*.docker.com"
*  issuer: C=US; O=GeoTrust Inc.; CN=RapidSSL SHA256 CA - G3
*  SSL certificate verify ok.
> HEAD /v2/repositories/library/mongo/manifests/3 HTTP/1.1
> Host: hub.docker.com
> User-Agent: curl/7.50.3
> Accept: */*
> 
< HTTP/1.1 404 NOT FOUND
HTTP/1.1 404 NOT FOUND
< Date: Fri, 04 Nov 2016 12:23:21 GMT
Date: Fri, 04 Nov 2016 12:23:21 GMT
< Content-Type: text/html; charset=utf-8
Content-Type: text/html; charset=utf-8
< X-Frame-Options: deny
X-Frame-Options: deny
< Strict-Transport-Security: max-age=31536000
Strict-Transport-Security: max-age=31536000
< Server: nginx
Server: nginx
< X-Content-Type-Options: nosniff
X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
X-XSS-Protection: 1; mode=block
< Strict-Transport-Security: max-age=31536000
Strict-Transport-Security: max-age=31536000

< 
* Curl_http_done: called premature == 0
* Connection #0 to host hub.docker.com left intact
```

Getting info on a specific tag is hard.